lib/drivers.js

```diff
@@ -0,0 +1,242 @@
+import path from 'path';
+import _ from 'lodash';
+import YAML from 'yaml';
+import log from './logger';
+import { fs, mkdirp } from 'appium-support';
+
+const DRIVERS_YAML = 'drivers.yaml';
+const DRIVERS_SCHEMA_REV = 1;
+
+// On Appium launch, figure out which drivers are installed
+let INSTALLED_DRIVERS = {};
+
+// This is a map of driver names to npm packages representing those drivers.
+// The drivers in this list will be available to the CLI so users can just
+// type 'appium driver install 'name'', rather than having to specify the full
+// npm package. I.e., these are the officially recognized drivers.
+const KNOWN_DRIVERS = {
+  uiautomator2: 'appium-uiautomator2-driver',
+  xcuitest: 'appium-xcuitest-driver',
+  youiengine: 'appium-youiengine-driver',
+  windows: 'appium-windows-driver',
+  mac: 'appium-mac-driver',
+  espresso: 'appium-espresso-driver',
+  tizen: 'appium-tizen-driver',
+};
+
+const INSTALL_TYPE_NPM = 'npm';
```

Comment:
this could be a map

Reply:
i am not sure how this would improve the situation? the reason for making it a constant is to not have to worry about the particular string value that we use in the cli to refer to 'npm' type installs.